### PR TITLE
Round the clip dimensions

### LINKED ISSUE #873: Round the clip dimensions
https://github.com/kblok/puppeteer/pull/15/commits/4e1e2fb7016b4a5b64f9c6fea524621ba21c2817

### PR DESCRIPTION
**Pull request by Meir017**
closes #873

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/ElementHandleTests/ScreenshotTests.cs [hunk 1]
```diff
@@ -150,7 +150,25 @@ public async Task ShouldNotHangWithZeroWidthHeightElement()
...
             var elementHandle = await Page.QuerySelectorAsync("div");
             var screenshot = await elementHandle.ScreenshotDataAsync();
-            Assert.NotNull(screenshot); 
+            Assert.NotNull(screenshot);
+        }
+
+        [Fact]
+        public async Task ShouldWorkForAnElementWithFractionalDimensions()
+        {
+            await Page.SetContentAsync("<div style=\"width:48.52px;height:19.8px;border:1px solid black;\"></div>");
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/ElementHandleTests/ScreenshotTests.cs]: notice this is 48.52 and not 48.51 like puppeteer
for some reason when using 48.51 it was parsed as 48.50 and `Math.Round` rounds it to 48 and not 49

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -1721,6 +1717,21 @@ private async Task<string> PerformScreenshot(ScreenshotType type, ScreenshotOpti
...
+        private Clip ProcessClip(Clip clip)
+        {
+            var x = Math.Round(clip.X);
+            var y = Math.Round(clip.Y);
+            
+            return new Clip
+            {
+                X = x,
+                Y = y,
+                Width = Math.Round(clip.Width + clip.X - x),
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs]:
```suggestion
                Width = Math.Round(clip.Width + clip.X - x, MidpointRounding.AwayFromZero),
```

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -1721,6 +1717,21 @@ private async Task<string> PerformScreenshot(ScreenshotType type, ScreenshotOpti
...
+                Height = Math.Round(clip.Height + clip.Y - y),
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs]:
```suggestion
                Height = Math.Round(clip.Height + clip.Y - y, MidpointRounding.AwayFromZero),
```

File: lib/PuppeteerSharp.Tests/ElementHandleTests/ScreenshotTests.cs (hunk 1 above)
Comment (kblok) [inline lib/PuppeteerSharp.Tests/ElementHandleTests/ScreenshotTests.cs]:
```suggestion
            await Page.SetContentAsync("<div style=\"width:48.51px;height:19.8px;border:1px solid black;\"></div>");
```

**Comment by kblok** [comment]
@Meir017 conflicted :(

**Comment by kblok** [comment]
Did you see my suggestions @Meir017 ?

**Comment by Meir017** [comment]
@kblok I accepted them :)

**Pull request merged**